src/Catel.MVVM/Catel.MVVM.Shared/MVVM/ViewModels/ViewModelMetadata.cs

```diff
@@ -25,8 +25,8 @@ internal class ViewModelMetadata
         /// <exception cref="ArgumentNullException">The <paramref name="models"/> is <c>null</c>.</exception>
         /// <exception cref="ArgumentNullException">The <paramref name="mappings"/> is <c>null</c>.</exception>
         /// <exception cref="ArgumentNullException">The <paramref name="validations"/> is <c>null</c>.</exception>
-        internal ViewModelMetadata(Type viewModelType, Dictionary<string, ModelInfo> models, Dictionary<string, ViewModelToModelMapping> mappings,
-            Dictionary<string, ValidationToViewModelAttribute> validations)
+        internal ViewModelMetadata(Type viewModelType, IReadOnlyDictionary<string, ModelInfo> models, IReadOnlyDictionary<string, ViewModelToModelMapping> mappings,
```

Comment:
Using dictionary interfaces might bring in some surprises. See http://mattwarren.org/2016/02/17/adventures-in-benchmarking-memory-allocations/ and check out the IDictionary part